packages/grafana-data/src/transformations/fieldReducer.ts

```diff
@@ -418,7 +418,7 @@ function calculateFirstNotNull(field: Field, ignoreNulls: boolean, nullAsZero: b
   const data = field.values;
   for (let idx = 0; idx < data.length; idx++) {
     const v = data[idx];
-    if (v != null && v !== undefined) {
+    if (v != null && v !== undefined && !isNaN(v)) {
```

Comment:
probably `if (v != null && !Number.isNaN(v))`

Reply:
ah good catch - `Number.isNaN` doesn't infer that value is `number` and instead is `unknown`

Currently working on adding a test and the [`reduceFields` function](https://github.com/grafana/grafana/blob/2ec3d3902186f3117e9796510cc303f194d0b615/packages/grafana-data/src/transformations/fieldReducer.ts#L57-L58) isn't playing nicely and still is returning `NaN` instead of the expected valid value in the test suite 🙃